src/lambda_handler.py

```diff
@@ -34,54 +164,253 @@ def task_default(event):
     return result
 
 
-def task_cowsay(event) -> dict[str, Union[Optional[str], bool]]:
-    """Generate an output message using the provided information."""
-    result: dict[str, Union[Optional[str], bool]] = {"message": None, "success": True}
+class FileConfig(TypedDict):
+    """Define the type structure of the dictionary in the file_configs variable."""
 
-    character: str = event.get("character", "tux")
-    if character not in cowsay.characters.CHARS.keys():
-        error_msg = 'Character "%s" is not valid.'
-        logging.error(error_msg, character)
-        failed_task(result, error_msg % character)
+    app_regex: re.Pattern[str]
+    description: str
+    filename: str
+    # I'd prefer to define ip_set as Set[Union[IPv4Address, IPv6Address]],
+    # but ipaddress.collapse_addresses() uses the type variable "_N" while
+    # ip_network returns Union[IPv4Network, IPv6network] which causes
+    # the mypy pre-commit hook to throw this error:
+    #   Value of type variable "_N" of "collapse_addresses" cannot be
+    #   "Union[IPv4Network, IPv6Network]"
+    # My solution to this problem is to simply define ip_set as Set[Any].
+    # For a similar issue and discussion, see
+    # https://github.com/python/typeshed/issues/2080
+    ip_set: Set[Any]
+    static_ips: List[str]
+
+
+def validate_event_data(
+    event: Dict[str, Any]
+) -> Tuple[Dict[str, Any], bool, List[str]]:
+    """Validate the event data and return a tuple containing the validated event, a boolean result (True if valid, False if invalid), and a list of error message strings."""
+    result = True
+    errors = []
+
+    # Account ID checks
+    if "account_ids" not in event:
+        errors.append('Missing required key "account_ids" in event.')
+    elif event["account_ids"] == []:
```

Comment:
I personally prefer the truthy check for empty lists.

```suggestion
    elif not event["account_ids"]:
```